python/oneflow/nn/functional/functional_deform_conv.py

```diff
@@ -0,0 +1,92 @@
+"""
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+from typing import Optional, Tuple, Union
+
+import oneflow as flow
+from oneflow.framework.tensor import Tensor
+
+
+def deform_conv2d(
+    input: Tensor,
+    offset: Tensor,
+    weight: Tensor,
+    bias: Optional[Tensor] = None,
+    stride: Tuple[int, int] = (1, 1),
+    padding: Tuple[int, int] = (0, 0),
+    dilation: Tuple[int, int] = (1, 1),
+    mask: Optional[Tensor] = None,
+) -> Tensor:
+
+    use_mask = mask is not None
+
+    if mask is None:
+        mask = flow.zeros((input.shape[0], 0), dtype=input.dtype).to(input.device)
+
+    stride_h = stride[0]
+    stride_w = stride[1]
+    pad_h = padding[0]
+    pad_w = padding[1]
+    dil_h = dilation[0]
+    dil_w = dilation[1]
+    weights_h, weights_w = weight.shape[-2:]
+
+    # TODO(yzm):Support rectangle convolution
+    # NOTE:Delete the following error reporting program after support rectangle convolution
+    if weights_h != weights_w:
+        raise RuntimeError(
+            "Rectangle convolution is not currently supported, please try square convolution"
+        )
+    if bias is not None:
+        if len(bias.shape) != 1 or bias.shape[0] != weight.shape[0]:
+            raise RuntimeError("invalid bias shape:got:" f"{bias.shape}")
```

Comment:
```suggestion
            raise RuntimeError(f"invalid bias shape: {bias.shape}")
```

这里的报错和 pytorch 是对齐的吗？这个 `:got:` 比较奇怪

Reply:
这个与pytorch确认了，对应的检查不在这个算子里面；已删除